JAXB (especially javax.xml.bind.DatatypeConverter) is not part of Java 9 by default

I already posted this to the [mailing list](https://groups.google.com/forum/?fromgroups#!topic/pac4j-dev/TTrazvqWD04), but here again for completeness:

Hi,

running the [undertow-pac4j-demo](https://github.com/pac4j/undertow-pac4j-demo) on Java 9 leads to this stacktrace:

```
org.pac4j.core.exception.TechnicalException: java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
    at org.pac4j.core.engine.DefaultSecurityLogic.perform(DefaultSecurityLogic.java:167) ~[pac4j-core-1.9.4.jar:na]
    at org.pac4j.undertow.handler.SecurityHandler.handleRequest(SecurityHandler.java:87) ~[undertow-pac4j-1.2.3.jar:na]
    at io.undertow.server.Connectors.executeRootHandler(Connectors.java:202) ~[undertow-core-1.3.23.Final.jar:1.3.23.Final]
    at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:802) ~[undertow-core-1.3.23.Final.jar:1.3.23.Final]
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) ~[na:na]
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) ~[na:na]
    at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
    at org.pac4j.core.util.JavaSerializationHelper.serializeToBase64(JavaSerializationHelper.java:30) ~[pac4j-core-1.9.4.jar:na]
    at org.pac4j.undertow.context.UndertowWebContext.setRequestAttribute(UndertowWebContext.java:165) ~[undertow-pac4j-1.2.3.jar:na]
    at org.pac4j.core.profile.ProfileManager.save(ProfileManager.java:113) ~[pac4j-core-1.9.4.jar:na]
    at org.pac4j.core.engine.DefaultSecurityLogic.perform(DefaultSecurityLogic.java:120) ~[pac4j-core-1.9.4.jar:na]
    ... 6 common frames omitted
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
    at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:466) ~[na:na]
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:563) ~[na:na]
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:496) ~[na:na]
    ... 10 common frames omitted
```

A temporary workaround is to start the JVM with --add-modules java.xml.bind, but that will go away in the future as well.
Since only javax.xml.bind.DatatypeConverter is used in pac4j-core, I would recommend switching to java.util.Base64 which was added in Java 8.

I'll submit a PR for this!

--- a/pac4j-cas/src/main/java/org/pac4j/cas/credentials/extractor/TicketAndLogoutRequestExtractor.java
+++ b/pac4j-cas/src/main/java/org/pac4j/cas/credentials/extractor/TicketAndLogoutRequestExtractor.java
@@ -1,5 +1,6 @@
 package org.pac4j.cas.credentials.extractor;
 
+import java.util.Base64;
 import org.jasig.cas.client.util.CommonUtils;
 import org.pac4j.cas.config.CasConfiguration;
 import org.pac4j.cas.logout.CasLogoutHandler;
@@ -14,7 +15,6 @@ import org.pac4j.core.util.CommonHelper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.xml.bind.DatatypeConverter;
 import java.util.zip.Inflater;
 
 /**
@@ -98,7 +98,7 @@ public class TicketAndLogoutRequestExtractor implements CredentialsExtractor<Tok
     }
 
     private String uncompressLogoutMessage(final String originalMessage) {
-        final byte[] binaryMessage = DatatypeConverter.parseBase64Binary(originalMessage);
+        final byte[] binaryMessage = Base64.getMimeDecoder().decode(originalMessage);
 
         Inflater decompresser = null;
         try {

--- a/pac4j-cas/src/test/java/org/pac4j/cas/client/CasClientTests.java
+++ b/pac4j-cas/src/test/java/org/pac4j/cas/client/CasClientTests.java
@@ -11,8 +11,8 @@ import org.pac4j.core.http.url.UrlResolver;
 import org.pac4j.core.util.TestsConstants;
 import org.pac4j.core.util.TestsHelper;
 
-import javax.xml.bind.DatatypeConverter;
 import java.nio.charset.StandardCharsets;
+import java.util.Base64;
 import java.util.zip.Deflater;
 
 import static org.junit.Assert.*;
@@ -173,7 +173,7 @@ public final class CasClientTests implements TestsConstants {
         final int resultSize = deflater.deflate(buffer);
         final byte[] output = new byte[resultSize];
         System.arraycopy(buffer, 0, output, 0, resultSize);
-        return DatatypeConverter.printBase64Binary(output);
+        return Base64.getEncoder().encodeToString(output);
     }
 
     @Test

--- a/pac4j-core/src/main/java/org/pac4j/core/util/JavaSerializationHelper.java
+++ b/pac4j-core/src/main/java/org/pac4j/core/util/JavaSerializationHelper.java
@@ -3,10 +3,10 @@ package org.pac4j.core.util;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.xml.bind.DatatypeConverter;
 import java.io.*;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Base64;
 import java.util.List;
 
 /**
@@ -34,7 +34,7 @@ public class JavaSerializationHelper {
      * @return the base64 string of the serialized object
      */
     public String serializeToBase64(final Serializable o) {
-        return DatatypeConverter.printBase64Binary(serializeToBytes(o));
+        return Base64.getEncoder().encodeToString(serializeToBytes(o));
     }
 
     /**
@@ -63,7 +63,7 @@ public class JavaSerializationHelper {
      * @return the unserialized Java object
      */
     public Serializable unserializeFromBase64(final String base64) {
-        return unserializeFromBytes(DatatypeConverter.parseBase64Binary(base64));
+        return unserializeFromBytes(Base64.getDecoder().decode(base64));
     }
 
     /**

